### PR TITLE
[State API] Add worker startup & initialization time to state API + use it for many_tasks 

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This PR adds information to retrieve
1. worker startup time
2. worker initialization time

from `ray list workers --detail`.

We also add summarization & print the result to many_tasks which will help debugging worker-related regressions. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by rickyyx — APPROVED** [review]
LGTM!
Should we make a perf metric out of it if this is being reliable and consistent?

**Comment by scv119** [comment]
looks we did a lot of plumbing.. is it really necessary?

**Comment by rkooo567** [comment]
Are you asking a feature is necessary or the plumbing is necessary?

Unfortunately plumbing is necessary to implement a feature

**Review by scv119 — COMMENTED** [review]
comments

File: src/ray/raylet/worker_pool.cc [hunk 1]
```diff
@@ -318,6 +318,8 @@ WorkerPool::BuildProcessCommandArgs(const Language &language,
   if (language == Language::PYTHON) {
     worker_command_args.push_back("--startup-token=" +
                                   std::to_string(worker_startup_token_counter_));
+    worker_command_args.push_back("--worker-launch-time-ms=" +
```
[the remark this answers is not in the document]
Comment (rkooo567) [reply src/ray/raylet/worker_pool.cc:321]: hmm we anyway need plumbing to calculate the initialization time although we use metrics
also metrics has cardinatliy issue + not practical at this point (until we have a default core dashboard, which we don't plan to have it for a while).

Having these values to worker state also makes sense given that's the direction we are going (add event information to state API)

**Review by scv119 — APPROVED** [review]
talked offline. while i don't think this is necessary but ok to move forward as it is for now.

File: src/ray/protobuf/gcs.proto [hunk 2]
```diff
@@ -378,15 +378,26 @@ message WorkerTableData {
...
+  */
+
+  // The unix ms timestamp the worker was started and finished initialization.
   uint64 start_time_ms = 23;
   // The unix ms timestamp the worker was ended at.
   uint64 end_time_ms = 24;
-  // The time when this worker is launched.
+  // The time when this worker process is requested from raylet.
+  // The field exists only when the worker is launched
+  // by a raylet. (I.e., driver worker won't have this value).
```
Comment (rickyyx) [inline src/ray/protobuf/gcs.proto:395]: why did we un-optional the fields?

**Review by rickyyx — APPROVED** [review]

File: src/ray/protobuf/gcs.proto (hunk 2 above)
Comment (rkooo567) [reply src/ray/protobuf/gcs.proto:395]: I added `// If the value doesn't present, it is -1.`!

File: python/ray/experimental/state/common.py [hunk 3]
```diff
@@ -488,10 +488,23 @@ class WorkerState(StateSchema):
...
+    #: worker_launch_time_ms (process startup requested).
+    #: -> worker_launched_time_ms (process started).
+    #: -> start_time_ms (worker is ready to be used).
+    #: -> end_time_ms (worker is destroyed).
+    worker_launch_time_ms: int = state_column(filterable=False, detail=True)
+    #: The time worker is succesfully launched
+    #: -1 if the value doesn't exist.
+    worker_launched_time_ms: int = state_column(filterable=False, detail=True)
     #: The time when the worker is started and initialized.
+    #: 0 if the value doesn't exist.
```
Comment (rickyyx) [inline python/ray/experimental/state/common.py:503]: should we also -1 default it?

**Comment by rickyyx** [comment]
should we have a release test run to see how this looks like?

**Review by rickyyx — APPROVED** [review]

File: python/ray/experimental/state/common.py (hunk 3 above)
Comment (rkooo567) [reply python/ray/experimental/state/common.py:503]: let me try

**Comment by rkooo567** [comment]
release test here https://buildkite.com/ray-project/release-tests-pr/builds/32132#01870e86-78ba-4ac8-b459-432d6105a00d

**Comment by rickyyx** [comment]
Somehow not seeing th eoutputs there in the release test?

**Pull request merged**